service/src/main/java/datawave/microservice/query/QueryManagementService.java

```diff
@@ -0,0 +1,2602 @@
+     * @return a generic response containing the new query id
+     * @throws NotFoundQueryException
+     *             if the query cannot be found
+     * @throws UnauthorizedQueryException
+     *             if the user doesn't own the query
+     * @throws QueryException
+     *             if query lock acquisition fails
+     * @throws BadRequestQueryException
+     *             if parameter validation fails
+     * @throws BadRequestQueryException
+     *             if query logic parameter validation fails
+     * @throws UnauthorizedQueryException
+     *             if the user doesn't have access to the requested query logic
+     * @throws BadRequestQueryException
+     *             if security marking validation fails
+     * @throws BadRequestQueryException
+     *             if auditing fails
+     * @throws QueryException
+     *             if query storage fails
+     */
+    private TaskKey duplicate(QueryStatus queryStatus, MultiValueMap<String,String> parameters, DatawaveUserDetails currentUser) throws QueryException {
+        // recreate the query parameters
+        MultiValueMap<String,String> currentParams = new LinkedMultiValueMap<>(queryStatus.getQuery().toMap());
+        
+        // remove some of the copied params
+        currentParams.remove(QUERY_ID);
+        currentParams.remove(USER_DN);
+        currentParams.remove(DN_LIST);
+        
+        // updated all of the passed in parameters
+        updateParameters(parameters, currentParams);
+        
+        // define a duplicate query
+        return storeQuery(currentParams.getFirst(QUERY_LOGIC_NAME), currentParams, queryStatus.getQueryKey().getQueryPool(), currentUser, CREATE);
+    }
+    
+    private boolean updateParameters(MultiValueMap<String,String> newParameters, MultiValueMap<String,String> currentParams) throws BadRequestQueryException {
+        return updateParameters(new ArrayList<>(newParameters.keySet()), newParameters, currentParams);
+    }
+    
+    /**
+     * Updates the current parameters with the new parameters, limited to the specified parameter names.
+     *
+     * @param parameterNames
+     *            the parameters names to override, not null
+     * @param newParameters
+     *            the new parameters, may be null
+     * @param currentParams
+     *            the current parameters, not null
+     * @return true is the current parameters were changed
+     * @throws BadRequestQueryException
+     *             if a parameter is passed without a value
+     */
+    private boolean updateParameters(Collection<String> parameterNames, MultiValueMap<String,String> newParameters, MultiValueMap<String,String> currentParams)
+                    throws BadRequestQueryException {
+        boolean paramsUpdated = false;
+        for (String paramName : parameterNames) {
+            if (newParameters.get(paramName) != null && !newParameters.get(paramName).isEmpty()) {
```

Comment:
Consider using `StringUtils.isNotBlank(newParameters.get(paramName))`

Reply:
`newParameters.get(paramName)` returns a list, not a string.  Is there an equivalent call for lists?

Reply:
figured it out.  should be better on both accounts now.